pkg/templates/kubeadm/v1beta3/kubeadm.go

```diff
@@ -375,10 +400,22 @@ func newNodeIP(host kubeoneapi.HostConfig) string {
 
 func newNodeRegistration(s *state.State, host kubeoneapi.HostConfig) kubeadmv1beta3.NodeRegistrationOptions {
 	kubeletCLIFlags := map[string]string{
-		"node-ip":           newNodeIP(host),
 		"volume-plugin-dir": "/var/lib/kubelet/volumeplugins",
 	}
 
+	if s.Cluster.ClusterNetwork.IPFamily == kubeoneapi.DualStack {
+		// If external or in-tree CCM is in use we don't need to set --node-ip
+		// as the cloud provider will know what IPs to return.
+		if !(s.Cluster.CloudProvider.External || s.Cluster.CloudProvider.None == nil) {
+			// TODO: shouldn't these be default interface ips?
```

Comment:
forgotten TODO?

Reply:
More of an unanswered question. K1 doesn't do it. Maybe it should?! :thinking:

Reply:
No, we explicitly don't use the default IP address provided by the operating system, but instead determine the default IP address ourselves. That's because you (almost) always want to use the private IP address for Kubelet (and Kubernetes in general). However, on some providers (such as Hetzner and DigitalOcean), the default IP address is a public address instead of a private one.

For example, we had something similar reported as an issue in OSM/MC recently: https://github.com/kubermatic/operating-system-manager/issues/230

Reply:
Keeping as it is.